src/indra/io/upload.py

```
@@ -88,3 +88,42 @@ def upload_data_to_s3(*,
+    :param str key:
+        S3 object key (full path within the bucket).
+    :param bool delete_after:
+        If ``True`` (default), delete the local file after a successful upload.
+    :returns:
+        ``True`` on success, ``False`` on failure (error already logged).
+    """
+    client = boto3.client("s3")
+    try:
+        client.upload_file(Filename=local_path, Bucket=bucket, Key=key)
+        logger.info("Uploaded %s → s3://%s/%s", os.path.basename(local_path), bucket, key)
+    except Exception as exc:
```

Ruff (0.15.9)

[warning] Do not catch blind exception: `Exception`

(BLE001)